pkg/query-service/app/logs/v4/resource_query_builder.go

```diff
@@ -0,0 +1,204 @@
+package v4
+
+import (
+	"fmt"
+	"strings"
+
+	v3 "go.signoz.io/signoz/pkg/query-service/model/v3"
+	"go.signoz.io/signoz/pkg/query-service/utils"
+)
+
+// buildResourceFilter builds a clickhouse filter string for resource labels
+func buildResourceFilter(logsOp string, key string, op v3.FilterOperator, value interface{}) string {
+	// we are using lower(labels) as we want case insensitive filtering
+	searchKey := fmt.Sprintf("simpleJSONExtractString(lower(labels), '%s')", key)
+
+	chFmtVal := utils.ClickHouseFormattedValue(value)
+
+	switch op {
+	case v3.FilterOperatorExists:
+		return fmt.Sprintf("simpleJSONHas(lower(labels), '%s')", key)
+	case v3.FilterOperatorNotExists:
+		return fmt.Sprintf("not simpleJSONHas(lower(labels), '%s')", key)
+	case v3.FilterOperatorRegex, v3.FilterOperatorNotRegex:
+		return fmt.Sprintf(logsOp, searchKey, chFmtVal)
```

Comment:
the existing order is correct, here is the example
logOp = match(%s, %s)
searchKey = simpleJSONExtractString(lower(labels), 'service_name')
chFmtVal = ".*'

finally it will be match(simpleJSONExtractString(lower(labels), 'service_name'), '.*')